File: apps/trustim-investigation/workbench/server-py/app/models/skill.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class Skill(BaseModel):
    name: str
    description: str
    allowed_tools: list[str] = Field(default_factory=list)
    file_path: str
    category: Literal["investigation", "action"]
    area: str


class SkillDetail(Skill):
    """Skill with full markdown content included."""
    content: str


class SkillInventory(BaseModel):
    investigation: list[Skill] = Field(default_factory=list)
    action: list[Skill] = Field(default_factory=list)

File: apps/trustim-investigation/workbench/src/types/skill.ts
/** Skill metadata parsed from SKILL.md frontmatter */
export interface Skill {
  name: string
  description: string
  allowed_tools: string[]
  file_path: string
  /** Derived category from directory structure */
  category: 'investigation' | 'action'
  /** Attack vector or functional area */
  area: string
}

/** Skills organized by category for the browser */
export interface SkillInventory {
  investigation: Skill[]
  action: Skill[]
}

/** Trace event for the log view (R36-R40) */
export interface TraceEvent {
  id: string
  timestamp: string
  type: 'tool_call' | 'skill_invocation' | 'query' | 'enrichment' | 'reasoning' | 'error'
  node_id: string | null
  summary: string
  detail: string
  duration_ms: number
  success: boolean
}

/** Graph view mode */
export type ViewMode = 'graph' | 'log' | 'heatmap' | 'timeline'
